lib/characters.coffee

```diff
@@ -103,36 +122,47 @@ class Characters
           char[side].splice key, 1
       return char
 
+  # Find the index of a character
+  # parma
+  indexOf: (character) ->
+    for char, index in @_model.val()
+      if char == character
+        return index
+    return -1
   # Apply a delta and return the new position
+  # @param delta [Object] a delta (see ot-types for more info)
+  # @param position [Integer] the position where to start applying the delta, defaults to 0
+  #
+  # @return [Integer] the position of the cursor after parsing the delta
   delta: (delta, position) ->
     if delta?
       if not position?
         position = 0
+
+      arentNull = (el) ->
+        el != null
+      if _.all delta.attributes, arentNull
+        operation = @_selections.select
+      else
+        operation = @_selections.unselect
+
       if delta.insert?
-        deltaSelection =
-          from: position
-          to: position + delta.insert.length
-          action: "set"
         @insert position, delta.insert
+        from = @val position
+        to = @val (position + delta.insert.length)
+        operation from, to, delta.attributes
```

Comment:
select & unselect expect to be called on `@_selection`. This won't work, because `this`, will refer to `window`.

Change the code or do:
```
operation.call @_selections, from, to, delta.attributes
```